spec/models/export/others/extract_spec.rb

```diff
@@ -0,0 +1,47 @@
+require 'rails_helper'
+
+RSpec.describe Export::Others::Extract do
+  describe '.all_relevant' do
+    subject(:all_relevant) { Export::Others::Extract.all_relevant }
+
+    context 'there are some complete and incomplete submissions, all with valid entries' do
+      let!(:complete_submission) do
+        create :submission_with_validated_entries, aasm_state: 'completed'
+      end
+      let!(:pending_submission) do
+        create :submission_with_validated_entries
+      end
```

Comment:
I suspect that `pending_submission` doesn't have any `other` entries and would hence pass without the date range.